pyproject.toml

```diff
@@ -236,7 +236,15 @@ pre-commit = [
   'pre-commit~=3.5',
   'sqlalchemy[mypy]~=2.0',
   'tomli',
-  'types-PyYAML'
+  'types-click-spinner~=0.1.8',
```

Comment:
Thanks @danielhollas, always love more typing! 😍 Just one question before I approve: could you motivate the choice of these version specifications?